dodiscover/context.py

```diff
@@ -112,3 +120,21 @@ def excluded_edges(self) -> nx.Graph:
     @property
     def init_graph(self) -> GraphProtocol:
         return self._init_graph
+
+    def copy(self):
+        """Create a copy of the Context object.
+
+        Returns
+        -------
+        context : Context
+            A copy.
+        """
+        context = Context(
+            data=copy(self._data),
+            variables=copy(self._variables),
+            latents=copy(self._latents),
+            init_graph=self._init_graph.copy(),
```

Comment:
This will produce a shallow copy of the graph.  Would need `copy.deepcopy` otherwise

Reply:
Changed to deepcopy